sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java

```diff
@@ -1228,52 +1517,94 @@ public void onTransportConnected(final TransportType type){
     		//If we have clients
-			notifyClients(createHardwareConnectedMessage(type));
+			notifyClients(createHardwareConnectedMessage(record));
     	}
 	}
 	
-	private Message createHardwareConnectedMessage(final TransportType type){
+	private Message createHardwareConnectedMessage(final TransportRecord record){
 			Message message = Message.obtain();
 			message.what = TransportConstants.HARDWARE_CONNECTION_EVENT;
+			message.arg1 = TransportConstants.HARDWARE_CONNECTION_EVENT_CONNECTED;
 			Bundle bundle = new Bundle();
-			bundle.putString(TransportConstants.HARDWARE_CONNECTED, type.name());
-    		if(MultiplexBluetoothTransport.currentlyConnectedDevice!=null){
+			bundle.putString(TransportConstants.HARDWARE_CONNECTED, record.getType().name());
+			bundle.putParcelableArrayList(TransportConstants.CURRENT_HARDWARE_CONNECTED, getConnectedTransports());
+
+			if(MultiplexBluetoothTransport.currentlyConnectedDevice!=null){
     			bundle.putString(CONNECTED_DEVICE_STRING_EXTRA_NAME, MultiplexBluetoothTransport.currentlyConnectedDevice);
     		}
+
 			message.setData(bundle);
 			return message;
 		
 	}
 	
-	public void onTransportDisconnected(TransportType type){
-		if(altTransportService!=null){  //If we still have an alt transport open, then we don't need to tell the clients to close
-			return;
-		}
-		Log.e(TAG, "Notifying client service of hardware disconnect.");
-		connectedTransportType = null;
-		isTransportConnected = false;
-		stopClientPings();
-		
-		exitForeground();//Leave our foreground state as we don't have a connection anymore
-		
-		if(packetWriteTaskMaster!=null){
-			packetWriteTaskMaster.close();
-			packetWriteTaskMaster = null;
-		}
-		
+	//public void onTransportDisconnected(TransportType type) {
+//TODO
+    //}
+	public void onTransportDisconnected(TransportRecord record){
 		cachedModuleVersion = -1; //Reset our cached version
 		if(registeredApps != null && !registeredApps.isEmpty()){
 			Message message = Message.obtain();
 			message.what = TransportConstants.HARDWARE_CONNECTION_EVENT;
+			message.arg1 = TransportConstants.HARDWARE_CONNECTION_EVENT_DISCONNECTED;
+
 			Bundle bundle = new Bundle();
-			bundle.putString(HARDWARE_DISCONNECTED, type.name());
+            bundle.putParcelable(TRANSPORT_DISCONNECTED, record);
+            //For legacy
+            bundle.putString(HARDWARE_DISCONNECTED, record.getType().name());
 			bundle.putBoolean(TransportConstants.ENABLE_LEGACY_MODE_EXTRA, legacyModeEnabled);
+
+			//Still connected transports
+			bundle.putParcelableArrayList(TransportConstants.CURRENT_HARDWARE_CONNECTED,getConnectedTransports());
+
 			message.setData(bundle);
 			notifyClients(message);
 		}
+		if(!getConnectedTransports().isEmpty()){
+			ArrayList<TransportRecord> transports = getConnectedTransports();
+			// Updates notification to one of still connected transport
+			enterForeground("Connected to " + transports.get(transports.size() - 1),0);
```

Comment:
The notification will be the first connected in the list of USB->TCP->BT, maybe not the one actively being used